src/library/columncache.cpp

```diff
@@ -13,6 +13,56 @@ const QString kSortNoCase = QStringLiteral("lower(%1)");
 const QString kSortNoCaseLex = mixxx::DbConnection::collateLexicographically(
         QStringLiteral("lower(%1)"));
 
+const QString kColumnNameByEnum[] = {
+        [ColumnCache::COLUMN_LIBRARYTABLE_ID] = LIBRARYTABLE_ID,
+        [ColumnCache::COLUMN_LIBRARYTABLE_ARTIST] = LIBRARYTABLE_ARTIST,
```

Comment:
this is unfortunately not valid C++.

Reply:
For my understanding this is the C++ 20 feature "designated initializers".

Reply:
:-( 
```
Error: /home/runner/work/mixxx/mixxx/src/library/columncache.cpp:27:9: error: array designators are a C99 extension [-Werror,-Wc99-designator]
        [ColumnCache::COLUMN_LIBRARYTABLE_ID] = {LIBRARYTABLE_ID,
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
``

Reply:
yup. Designated initializers are cool, but they deliberately work differently to allow for for example default arguments. I'm afraid there is no real way of achieving this trick in C++. You'll implicitly have to assume that the order of the list matches the enum order. The better strategy would instead simply be to not rely on the enum order at all though. Instead of having a bunch of globals arrays, consider having a single array of structs that bundles these together.

Reply:
> You'll implicitly have to assume that the order of the list matches the enum order. 

A appreciate exactly this behavior a feature. This makes sure the list is complete and consistent. 

I don't understand you proposal. 

I would like to treat this as a compiler feature we can use like `#pragma once`, and just silence clazy. 

What do you think?

Reply:
> I don't understand you proposal.

Well currently you have
```cpp
auto kColumnPropertiesByEnum[] = {}; 
auto kColumnPropertiesByEnum[] = {}; 

> I would like to treat this as a compiler feature we can use like #pragma once, and just silence clazy.

I'm not sure its worth the trouble trying to opt into this extension and maintain it on every compiler. I'm also not sure what the semantics are when the enum suddenly gets different values (are entries in the array that are now empty unitialized, zero or default initialized?) I don't think we should mix C99 extensions with C++ code. 

Have you instead considered putting this into a utility function with a switch-case? Performance is likely comparable and this time we don't have issues if the enum suddenly gets gaps and we can rely on `-Wswitch` for exhaustiveness (also we don't need `NUM_COLUMNS` anymore, which is always a hack).

Reply:
> I'm also not sure what the semantics are when the enum suddenly gets different values 

That is the great thing with this "feature". Compiling fails. So you cannot mess up the initalization you can do if this are single cals or an anonymous initalizer like in MSVC.

Reply:
Ok. I won't stress this for now then. Lets focus on the other stuff first.